tests/nvmeof/workflows/ns_masking.py

```diff
@@ -0,0 +1,384 @@
+import json
+import time
+from collections import defaultdict
+from concurrent.futures import ThreadPoolExecutor
+
+from ceph.ceph import CommandFailed
+from ceph.parallel import parallel
+from ceph.utils import get_node_by_id
+from tests.nvmeof.workflows.initiator import NVMeInitiator
+from utility.log import Log
+from utility.retry import retry
+
+LOG = Log(__name__)
+
+
+class NamespaceMasking:
+    """
+    Helper class containing the original methods.
+    """
+
+    def __init__(
+        self,
+        cluster,
+        gateways,
+        clients,
+        orch,
+    ):
+        self.cluster = cluster
+        self.gateways = gateways
+        self.clients = clients
+        self.orch = orch
+        self.initiators = {}
+
+        # Track executor and io_tasks state
+        self.executor = None
+        self.io_tasks = []
+
+    def get_or_create_initiator(self, node_id, nqn):
+        """Get existing NVMeInitiator or create a new one for each (node_id, nqn)."""
+        key = (node_id, nqn)  # Use both as dictionary key
+
+        if key not in self.initiators:
+            node = get_node_by_id(self.cluster, node_id)
+            self.initiators[key] = NVMeInitiator(node, nqn)
+
+        return self.initiators[key]
+
+    def prepare_io_execution(self, io_clients, return_clients=False):
```

Comment:
get_or_create_initiator , prepare_io_execution and validate_io are common methods for almost all our test modules, so these can be moved to initiator.py

Reply:
looks like this PR was not raised when I requested for review. Agreed. I shall  use methods form initiator.py in upcoming PR(as stated in PR description). For now, please review this despite of this. Looks like code files are bit messed up on my local when seen today